tests_integration/nativeapp/test_feature_metrics.py

```diff
@@ -0,0 +1,123 @@
+from shlex import split
+from typing import Dict
+from unittest import mock
+from unittest.mock import MagicMock
+
+from snowflake.cli.api.metrics import CLICounterField
+from tests.project.fixtures import *
+
+
+def _extract_first_result_executing_command_telemetry_message(
+    mock_telemetry: MagicMock,
+):
+    return next(
+        args.args[0].to_dict()["message"]
+        for args in mock_telemetry.call_args_list
+        if args.args[0].to_dict().get("message").get("type")
+        == "result_executing_command"
+    )
+
+
+@pytest.mark.integration
+@pytest.mark.parametrize(
+    "command,expected_counter",
+    [
+        (
+            [
+                "sql",
+                "-q",
+                "select '<% ctx.env.test %>'",
+                "--env",
+                "test=value_from_cli",
+            ],
+            1,
+        ),
+        (["sql", "-q", "select 'string'"], 0),
+    ],
+)
+@mock.patch("snowflake.connector.telemetry.TelemetryClient.try_add_log_to_batch")
+def test_sql_templating_emits_counter_one(
+    mock_telemetry,
+    command: List[str],
+    expected_counter,
+    runner,
+):
+    result = runner.invoke_with_connection_json(command)
+
+    assert result.exit_code == 0
+
+    message = _extract_first_result_executing_command_telemetry_message(mock_telemetry)
+
+    assert message["counters"][CLICounterField.SQL_TEMPLATES] == expected_counter
+
+
+@pytest.mark.integration
+@pytest.mark.parametrize(
+    "command," "test_project," "expected_counters",
+    [
+        (
+            "app deploy",
+            "napp_application_post_deploy_v1",
+            {CLICounterField.PDF_TEMPLATES: 0, CLICounterField.POST_DEPLOY_SCRIPTS: 1},
+        ),
+        (
+            "ws bundle --entity-id=pkg",
+            "napp_templates_processors_v2",
+            {
+                CLICounterField.SNOWPARK_PROCESSOR: 0,
+                CLICounterField.TEMPLATES_PROCESSOR: 1,
+                CLICounterField.PDF_TEMPLATES: 1,
+            },
+        ),
+        (
+            "app run",
+            "na_snowpark_processor",
+            {
+                CLICounterField.SNOWPARK_PROCESSOR: 1,
+                CLICounterField.TEMPLATES_PROCESSOR: 0,
+            },
+        ),
+    ],
+)
+@mock.patch("snowflake.connector.telemetry.TelemetryClient.try_add_log_to_batch")
+def test_nativeapp_feature_counter_has_expected_value(
+    mock_telemetry,
+    runner,
+    nativeapp_teardown,
+    nativeapp_project_directory,
+    command: str,
+    test_project: str,
+    expected_counters: Dict[str, int],
+):
+    local_test_env = {
+        "APP_DIR": "app",
+        "schema_name": "test_schema",
+        "table_name": "test_table",
+        "value": "test_value",
+    }
+
+    with nativeapp_project_directory(test_project):
+        runner.invoke_with_connection(split(command), env=local_test_env)
+
+        # The method is called with a TelemetryData type, so we cast it to dict for simpler comparison
+        message = next(
```

Comment:
can't we use `_extract_first_result_executing_command_telemetry_message` for this?

Reply:
Good catch, I extracted it from the other test and forgot to do it for this one too, thank you!